Sync develop branch with recent commits

ImJeremyHe opened the pull request:
Commits are cherry-picked from `celestia-integration` branch.

Currently this file test/foundry/ExpressLaneBalance.t.sol can't be compiled, and no workarounds have been found. Based on the fact that this file is only for test, I skip it in CI and `yarn:build`.

Relevant issues:

https://github.com/ethereum/solidity/issues/14179
https://github.com/ethereum/solidity/issues/14878

Sneh1999 committed:
Adding automata verifier to Celestia Contracts branch (#24)

* forge install: automata-dcap-attestation

v0.1.1

* Add automata attestation to Sequencer Inbox

* Add automata verifier to the Sequencer Inbox

* Update tests and contracts

* add address to the end

* make hardhat build work as well

* make tests work

* make tests work

* attempt tests

* add tests

* add tests for EspressoTEEVerifier

* add tests to SequencerInbox

* revert 4844 chanegs

* source .env

* fix broken ci

* fix test

* Make espressoTEEVerifier non upgrdeable and use V3QuoteVerifier contract directly

* address comments

Sneh1999 committed:
submodule update

Sneh1999 committed:
Update nitro scripts (#25)

* Update scripts to use EspressoTEEVerifier

* Update scripts to use EspressoTEEVerifier

* address comments

* check for undefined config

* address comments and fix typo

Sneh1999 committed:
Update SequencerInboxStub to fix tests (#26)

Sneh1999 committed:
Fix hardhat verification on arbSepolia (#29)

Sneh1999 committed:
Add additional verification to EspressoTEEVerifier (#28)

* Add additional verification to EspressoTEEVerifier

* fix tests

* fix tests

* cleanup

* revert broken tests

* fix ci

* fix ci

* address comments

* address comments

* add code docs

Sneh1999 reviewed (commented):
Thanks @ImJeremyHe  🎉 🎉  This looks good too but we should try to fix the build!

Sneh1999 reviewed (commented):
One more commit that we will have to cherry pick: https://github.com/EspressoSystems/nitro-contracts/pull/31

Sneh1999 committed:
Contracts cleanup (#31)

* Contracts cleanup

* add comment

ImJeremyHe committed:
Fix CI

ImJeremyHe committed:
Fix the test

In openzeppelin v4.8, revert message changed and error InvalidSignatureV is deprecated

ImJeremyHe committed:
Fix 4844 test

Sneh1999 committed:
run all ci steps

Sneh1999 committed:
remove unwanted variable

Sneh1999 committed:
revert CI changes

Sneh1999 reviewed (approved):
LGTM!

I was trying to see if we can fix the other skipped CI steps but it seems like a separate issue. We can fix it later
